src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -59,17 +64,58 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
             return await _scriptHostManager.HttpRequestManager.ProcessRequestAsync(request, processRequestHandler, cancellationToken);
         }
 
-        public static async Task<AuthorizationLevel> DetermineAuthorizationLevelAsync(HttpRequestMessage request, FunctionDescriptor function, IDependencyResolver resolver)
+        public static async Task<AuthorizationLevel> DetermineAuthorizationLevelAsync(HttpRequestMessage request, FunctionDescriptor function, ScriptSettingsManager settingsManager, IDependencyResolver resolver)
         {
+            // first check if we're authorized via key
             var secretManager = resolver.GetService<ISecretManager>();
             var authorizationResult = await AuthorizationLevelAttribute.GetAuthorizationResultAsync(request, secretManager, functionName: function.Name);
             var authorizationLevel = authorizationResult.AuthorizationLevel;
+
+            if (authorizationLevel > AuthorizationLevel.Anonymous)
+            {
+                // add a new identity to the principal representing
+                // the key authentication result
+                AddKeyAuthorizationIdentity(ClaimsPrincipal.Current, authorizationResult);
+            }
+
+            // See if we're we're authorized at the User level (EasyAuth)
```

Comment:
I'm still not sure that I like that assertion. If the developer specifies "user" then that's clearly the most important thing to them. To ignore that in favor of a key seems like a violation of expectation. The user token is also a stronger security claim, so I think we want to enforce it when we're asked to. The point at which a shared secret is going into a client that a user interacts with is generally the point at which it should no longer be considered a secret.

I think we'd also discussed splitting keys and user config settings to allow composite authorization checks. That would mean that "anonymous" would have to be renamed (arguably, it should be anyway).